### PR TITLE
Updated GuardDuty alert with more entities

### PR DESCRIPTION
**Pull request by petebryan**
Required items, please complete
   
   Change(s):
   - Updated GuardDuty alert template to extract more details.

   Reason for Change(s):
   - Query improvement

   Version Updated:
   - Y

   Testing Completed:
   - Y

   Checked that the validations are passing and have addressed any issues that are present:
   - Y

### CODE REVIEW

File: Solutions/Amazon Web Services/Analytic Rules/AWS_GuardDuty_template.yaml [hunk 1]
```diff
@@ -15,27 +15,95 @@ tactics: []
...
+    isnotempty(ServiceDetails.action.awsApiCallAction), ServiceDetails.action.awsApiCallAction,
+    isnotempty(ServiceDetails.action.dnsRequestAction), ServiceDetails.action.dnsRequestAction,
+    isnotempty(ServiceDetails.action.kubernetesApiCallAction), ServiceDetails.action.kubernetesApiCallAction,
+    isnotempty(ServiceDetails.action.networkConnectionAction), ServiceDetails.action.networkConnectionAction,
+    isnotempty(ServiceDetails.action.portProbeAction), ServiceDetails.action.portProbeAction,
+    isnotempty(ServiceDetails.action.rdsLoginAttemptAction), ServiceDetails.action.rdsLoginAttemptAction,
+    dynamic(null)
+    )
+  | extend RemoteIpAddress = max_of(
+    tostring(ServiceAction.remoteIpDetails.ipAddressV4),
```
Comment (devikamehra) [inline Solutions/Amazon Web Services/Analytic Rules/AWS_GuardDuty_template.yaml:59]: There is no entity with the same ServiceAction as everything was renamed as ActionDetailsObject in line 49 to 56.

File: Solutions/Amazon Web Services/Analytic Rules/AWS_GuardDuty_template.yaml [hunk 2]
```diff
@@ -15,27 +15,95 @@ tactics: []
...
+    tostring(parse_json(ServiceAction.sourceIPs)[0])
+    )
+  | extend AccountUpn = case(
+    AccessKeyDetails.userType == "IAMUser", AccessKeyDetails.userName,
```
Comment (devikamehra) [inline Solutions/Amazon Web Services/Analytic Rules/AWS_GuardDuty_template.yaml:63]: No field AccessKeyDetails or RdsDbUserDetails

**Review by devikamehra — CHANGES_REQUESTED** [review]

**Comment by v-atulyadav** [comment]
Hi @petebryan, 
This PR is in draft since quite a long. Please let us know if you need any help on this. 
Thanks!

File: Solutions/Amazon Web Services/Analytic Rules/AWS_GuardDuty_template.yaml [hunk 3]
```diff
@@ -13,29 +13,131 @@ triggerOperator: gt
...
+        columnName: UPNSuffix
       - identifier: ObjectGuid
-        columnName: AWSAcoundId
+        columnName: RemoteAWSAccountId
+  - entityType: IP
+    fieldMappings:
+      - identifier: Address
+        columnName: RemoteIpAddress
+      - identifier: Address
+        columnName: LocalIpAddress
```
Comment (devikamehra) [inline Solutions/Amazon Web Services/Analytic Rules/AWS_GuardDuty_template.yaml:133]: Two values are provided for same Identifier. We would have to create a new block for IP entity if we need to map two IPs

File: Solutions/Amazon Web Services/Analytic Rules/AWS_GuardDuty_template.yaml [hunk 4]
```diff
@@ -13,29 +13,131 @@ triggerOperator: gt
...
 customDetails:
+  UniqueFindingId: UniqueFindingId
```
Comment (devikamehra) [inline Solutions/Amazon Web Services/Analytic Rules/AWS_GuardDuty_template.yaml:135]: Field is not present in the schema

**Review by devikamehra — CHANGES_REQUESTED** [review]

**Comment by v-rusraut** [comment]
Hi @petebryan 
Please work on @devikamehra's comment.
Thanks

File: Solutions/Amazon Web Services/Analytic Rules/AWS_GuardDuty_template.yaml (hunk 3 above)
Comment (petebryan) [reply Solutions/Amazon Web Services/Analytic Rules/AWS_GuardDuty_template.yaml:135]: Fixed

**Comment by petebryan** [comment]
> Hi @petebryan Please work on @devikamehra's comment. Thanks

Done

**Review by devikamehra — DISMISSED** [review]
@petebryan Thank you for making the changes. I am good with the PR.
Just a small suggestion, can we move the version to the last line of the file?

**Comment by petebryan** [comment]
> @petebryan Thank you for making the changes. I am good with the PR. Just a small suggestion, can we move the version to the last line of the file?

Done - ill update the Wiki to reflect this requirement for others.

**Review by devikamehra — DISMISSED** [review]

**Comment by v-rusraut** [comment]
We are working on to fix failed validation checks.
We will provide update on 13-07-2023.
Thanks

**Review by v-atulyadav — APPROVED** [review]

**Pull request merged**